nautobot/extras/tests/test_api.py

```diff
@@ -1106,14 +1142,205 @@ def test_run_job_with_missing_data(self):
+        "name_override": True,
+        "name": "Overridden name",
+        "slug": "overridden-slug",
+        "description_override": True,
+        "description": "This is an overridden description.",
+        "enabled": True,
+        "approval_required_override": True,
+        "approval_required": True,
+        "commit_default_override": True,
+        "commit_default": False,
+        "hidden_override": True,
+        "hidden": True,
+        "read_only_override": True,
+        "read_only": True,
+        "soft_time_limit_override": True,
+        "soft_time_limit": 350.1,
+        "time_limit_override": True,
+        "time_limit": 650,
+    }
+    bulk_update_data = {
+        "enabled": True,
+        "approval_required_override": True,
+        "approval_required": True,
+    }
+    validation_excluded_fields = []
+
+    run_success_response_status = status.HTTP_201_CREATED
+
+    def setUp(self):
+        super().setUp()
+        job_model = Job.objects.get_for_class_path("local/api_test_job/APITestJob")
+        job_model.enabled = True
+        job_model.validated_save()
+
+    def get_run_url(self, class_path="local/api_test_job/APITestJob"):
+        job_model = Job.objects.get_for_class_path(class_path)
+        return reverse("extras-api:jobmodel-run", kwargs={"pk": job_model.pk})
+
+    def _get_detail_url(self, instance):
+        """
+        Override default implementation as we're under "jobmodel-detail" rather than "job-detail".
+        """
+        viewname = f"{self._get_view_namespace()}:jobmodel-detail"
+        return reverse(viewname, kwargs={"pk": instance.pk})
+
+    def _get_list_url(self):
+        """
+        Override default implementation as we're under "jobmodel-list" rather than "job-list".
+        """
+        viewname = f"{self._get_view_namespace()}:jobmodel-list"
+        return reverse(viewname)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_run_job_object_var(self):
+        """In addition to the base test case provided by JobAPIRunTestMixin, also verify the JSON response data."""
+        response, schedule = super().test_run_job_object_var()
+
+        self.assertIn("schedule", response.data)
+        self.assertIn("job_result", response.data)
+        self.assertEqual(response.data["schedule"], NestedScheduledJobSerializer(schedule).data)
+        self.assertIsNone(response.data["job_result"])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_run_job_object_var_lookup(self):
+        """In addition to the base test case provided by JobAPIRunTestMixin, also verify the JSON response data."""
+        response, job_result = super().test_run_job_object_var_lookup()
+
+        self.assertIn("schedule", response.data)
+        self.assertIn("job_result", response.data)
+        self.assertIsNone(response.data["schedule"])
+        # The urls in a NestedJobResultSerializer depends on the request context, which we don't have
```

Comment:
Could you fake a request with `django.test.RequestFactory`? This comment helps but it's not entirely clear why deleting the `url` fields is necessary otherwise. (It's not a big deal, just more of a curiosity.)

Reply:
I tried some various approaches using DRF's `APIRequestFactory` but couldn't figure out the magic sauce to make it work - I kept getting 403 Forbidden responses even when explicitly using `force_authenticate(request)`. Leaving this as-is for now.